e2e/src/usecases/DappListRecent.js

```diff
@@ -0,0 +1,158 @@
+import { reloadReactNative } from '../utils/retries'
+import { sleep, waitForElementId, getElementText, getElementTextList } from '../utils/utils'
+
+const jestExpect = require('expect')
+
+/**
+ * From the home screen, navigate to the dapp explorer screen
+ * Disable detox device synchronization to allow for dapp list to load
+ * https://github.com/wix/Detox/issues/2799
+ */
+async function navigateToDappList() {
+  await device.disableSynchronization()
+  await waitForElementId('Hamburger')
+  await element(by.id('Hamburger')).tap()
+  await waitForElementId('dapps-explorer-icon')
+  await element(by.id('dapps-explorer-icon')).tap()
+  await waitFor(element(by.id('DAppExplorerScreen/loading')))
+    .not.toBeVisible()
+    .withTimeout(30 * 1000)
+  await device.enableSynchronization()
+}
+
+/**
+ * From the drawer navigate to home screen
+ */
+async function navigateToHome() {
+  await waitForElementId('Hamburger')
+  await element(by.id('Hamburger')).tap()
+  await waitForElementId('Home')
+  await element(by.id('Home')).tap()
+}
+
+/**
+ * Scroll to a dapp in the dapp - iOS only
+ * @param {number} dappIndex: index of dapp to scroll to
+ */
+async function scrollToDapp(dappIndex = 0) {
+  try {
+    await waitFor(element(by.id('DappCard')).atIndex(dappIndex))
+      .toBeVisible(100)
+      .whileElement(by.id('DAppExplorerScreen/DappsList'))
+      .scroll(250, 'down')
+  } catch {
+    console.log('Catch of scrollToDapp')
+  }
+}
+
+export default DappListRecent = () => {
+  it('should show most recently used dapp leftmost on home screen :ios:', async () => {
+    // Get recently used dapps at start of test
+    const startRecentlyUsedDapps = await getElementTextList('RecentDapp-name')
+    const startRecentDappCount = startRecentlyUsedDapps.length
+
+    // If no recently used dapps check that RecentDapp element does not exist
+    // Else check that All Dapps button exists
```

Comment:
i was confused for a sec about why the test would need conditions (like, why isn't it deterministic?) but then i remembered you said that you wanted this test to be able to rerun from any state left by previous tests. could you also mention this in the comment so that we don't forget?

Reply:
Added a comment in d30b567a9d4522ff412edde8e618d5b1f284f183!